flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecLookupJoin.java

```diff
@@ -294,4 +355,47 @@ protected Transformation<RowData> createSyncLookupJoinWithState(
         transform.setStateKeyType(keySelector.getProducedType());
         return transform;
     }
+
+    private RowDataKeySelector getKeySelector(
+            boolean singleParallelism, int[] keys, ClassLoader classLoader, RowType inputRowType) {
+        RowDataKeySelector keySelector;
+        if (singleParallelism) {
+            keySelector = EmptyRowDataKeySelector.INSTANCE;
+        } else {
+            // make it a deterministic asc order
+            Arrays.sort(keys);
+            keySelector =
+                    KeySelectorUtil.getRowDataSelector(
+                            classLoader, keys, InternalTypeInfo.of(inputRowType));
+        }
+        return keySelector;
+    }
+
+    private Transformation<RowData> createPartitionTransformation(
+            RowDataKeySelector keySelector,
+            Transformation<RowData> inputTransformation,
+            boolean singleParallelism,
+            ExecNodeConfig config) {
+
+        final KeyGroupStreamPartitioner<RowData, RowData> partitioner =
+                new KeyGroupStreamPartitioner<>(
+                        keySelector, KeyGroupRangeAssignment.DEFAULT_LOWER_BOUND_MAX_PARALLELISM);
+        Transformation<RowData> partitionedTransform =
+                new PartitionTransformation<>(inputTransformation, partitioner);
+
+        createTransformationMeta(PARTITIONER_TRANSFORMATION, "Partitioner", "Partitioner", config)
```

Comment:
Why we import this field from `CommonExecSink`...? Can we use a new & same one in `StreamExecLookupJoin`?